src/groups/bmq/bmqp/bmqp_recoveryeventbuilder.cpp

```diff
@@ -39,23 +39,31 @@ namespace bmqp {
 // --------------------------
 
 // CREATORS
-RecoveryEventBuilder::RecoveryEventBuilder(
-    bdlbb::BlobBufferFactory* bufferFactory,
-    bslma::Allocator*         allocator)
-: d_blob(bufferFactory, allocator)
+RecoveryEventBuilder::RecoveryEventBuilder(BlobSpPool*       blobSpPool_p,
+                                           bslma::Allocator* allocator)
+: d_blobSpPool_p(blobSpPool_p)
+, d_blob_sp(0, allocator)  // initialized in `reset()`
+, d_emptyBlob_sp(blobSpPool_p->getObject())
```

Comment:
1. Removed `d_emptyBlob_sp` fields from event builders.
2. Still keep the check for `BlobBufferFactory` in constructor with a sample disposable `BlobSp` to avoid runtime costs.
3. Return `d_blob_sp` when no messages added.